src/middlewared/middlewared/plugins/zfs_/pool_utils.py

```diff
@@ -20,9 +20,20 @@ def convert_topology(zfs, vdevs):
             topology[vdev['root'].lower()].extend(children)
         else:
             z_vdev = libzfs.ZFSVdev(zfs, 'disk')
-            z_vdev.type = vdev['type'].lower()
             z_vdev.children = children
-            topology[vdev['root'].lower()].append(z_vdev)
+            if vdev['type'].startswith('DRAID'):
+                vdev_type = vdev['type'][:-1].lower()
```

Comment:
There is no reason to do `vdev_type = vdev['type'][:-1].lower()`. Just do `z_vdev.type = 'drain'` as the latter is much more readable.